app/uploaders/avatar_uploader.rb

```diff
@@ -3,6 +3,15 @@
 class AvatarUploader < CarrierWave::Uploader::Base
   include Cloudinary::CarrierWave
 
+  # Sets the user's avatar to the given image url if the user
+  # doesn't currently have an avatar.
+  def self.set_avatar_from_url!(user, url)
+    if !user.no_third_party_avatar && (!user.avatar.file || !user.avatar.file.exists?)
```

Comment:
I think you can replace `!user.avatar.file || !user.avatar.file.exists?` with `!user.avatar.file.blank?`